test/integration/imageimporter_test.go

```diff
@@ -547,16 +562,30 @@ func TestImageStreamImportQuayIO(t *testing.T) {
 	rt, _ := kclient.TransportFor(&kclient.Config{})
 	importCtx := importer.NewContext(rt, nil).WithCredentials(importer.NoCredentials)
 
+	repositoryName := quayRegistryName + "/coreos/etcd"
 	imports := &api.ImageStreamImport{
 		Spec: api.ImageStreamImportSpec{
 			Images: []api.ImageImportSpec{
-				{From: kapi.ObjectReference{Kind: "DockerImage", Name: "quay.io/coreos/etcd"}},
+				{From: kapi.ObjectReference{Kind: "DockerImage", Name: repositoryName}},
 			},
 		},
 	}
 
-	i := importer.NewImageStreamImporter(importCtx, 3, nil)
-	if err := i.Import(gocontext.Background(), imports); err != nil {
+	err := retryWhenUnreachable(t, func() error {
+		i := importer.NewImageStreamImporter(importCtx, 3, nil)
+		if err := i.Import(gocontext.Background(), imports); err != nil {
+			return err
+		}
+		d := imports.Status.Images[0]
+		if d.Status.Status != unversioned.StatusSuccess {
+			if d.Status.Reason == "NotV2Registry" {
```

Comment:
Quay.io periodically returns flaky responses. This is specific to quay
and should not be removed.